update gcp docs to make it clear how to limit infra hosts (#22272)
DataDog/documentation: content/en/cloud_cost_management/google_cloud.md

## Overview

> To use Google Cloud Cost Management in Datadog, you must: 

Reviewer: ```suggestion
To use Google Cloud Cost Management in Datadog, follow these steps: 
```

| Cost Type | Description |
| ----------| ----------------------------------|
| `gcp.cost.amortized` | Total cost of resources allocated at the time of usage over an interval. Costs include promotion credits as well as committed usage discount credits. |
| `gcp.cost.amortized.shared.resources.allocated` | All of your GCP amortized costs, with additional breakdowns and insights for container workloads. Requires [container cost allocation][14].|

### Container allocation

> **Container allocation** metrics contain all of the same costs as the GCP metrics, but with additional breakdowns and insights for container workloads. See [container cost allocation][14] for more details.

Reviewer: ```suggestion
**Container allocation** metrics contain all of the same costs as the Google Cloud Platform metrics, but with additional breakdowns and insights for container workloads. See [Container Cost Allocation][14] for more details.
```
